### ISSUE TITLE
[Machine Learning]Model Interpretation using LIME

### ISSUE THREAD

**Issue by willyngashu** [section-engineering-education/engineering-education#3970]
#### Proposed title of article
[Machine Learning]Model Interpretation using LIME

#### Proposed article introduction
Model interpretation involves finding out ways to understand model decision-making policies better. This is to enable fairness, accountability, and transparency which will give humans enough confidence to use these models in real-world problems which a lot of impact on business and society.

Model interpretation tries to answer the question why should I trust the model?. You should be able to explain what your model is doing. It is not enough to just trust your model prediction accuracy or score. Every software has bugs even so models can also have bugs(errors). Through interpretation, we should be able to find errors in the model and try to resolve these errors so that we can know how well our model learned during the training phase.

Regardless of what problem the model is solving an interpretable model will always be preferred because the end-users can understand what your model is really doing. Model Interpretability also helps you debug your model by giving you a chance to see what the model really thinks is important.
Furthermore, you can use interpretable models to combat the common belief that Machine Learning algorithms are black boxes and that we humans aren’t capable of gaining any insights into how they work.

An interpretable model helps you to understand and account for the factors that are not included in the model and account for the context of the problem when taking actions based on model predictions. Improving generalization and performance. High interpretability typically leads to a model that generalizes better.
Machine learning model fairness and interpretability are critical for data scientists, researchers, and developers to explain their models and understand the value and accuracy of their findings. Interpretability is also important to debug machine learning models and make informed decisions about how to improve them.

In this tutorial, we shall use LIME and Eli5 in interpreting our model. LIME  attempts to understand the model by perturbing the input of data samples and understanding how the predictions change. ELI5 is a python package that is used to inspect ML classifiers and explain their predictions. It is popularly used to debug algorithms such as sklearn regressors and classifiers.

#### Key takeaways 
1. What is model interpretation.
2.  Model building and training.
3. What is LIME and Eli5.
4. Interpreting our model using LIME and Eli5.

#### Article quality
This tutorial is unique since it builds a unique model from scratch, Breast Cancer Prediction, after building our model we then interpret our model using LIME and Eli5. This checks if our model was fully trained and we can be able to trust it. This is a practical approach that anyone can follow.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by lalith1403** [comment]
Great topic 🚀 , make sure it matches the following: 
* Adds value beyond what is in the official docs and what is covered in other blog sites.
* To avoid potential article closure, it should not overlap with existing EngEd articles or incoming topic suggestions.

Please reference any relevant EngEd articles in yours and build a unique project - Approved. @willyngashu
